### PR TITLE
Add CCR/CSS challenge to elastic/logs

### PR DESCRIPTION
**Pull request by b-deam**
This is a (very slightly) modified version of @dnhatn's previous work. For now, I think it makes sense to continue with the custom runners, but I opened https://github.com/elastic/rally/issues/1562 for tracking.

### CODE REVIEW

**Comment by b-deam** [comment]
I pushed which includes some logic to check if the remotes are actually connected, and that the licenses are valid.

Below are some test invocations;
```
### Cluster 1
# install with basic license to fail
esrally install --quiet --distribution-version=7.17.5 --node-name="cluster-1-node-1" --cluster-name="cluster-1" --network-host="127.0.0.1" --http-port=39210 --master-nodes="cluster-1-node-1" --seed-hosts="127.0.0.1:39300" | jq --raw-output '.["installation-id"]'
esrally start --installation-id $ID --race-id 1

# install with trial license to succeeed
esrally install --quiet --distribution-version=7.17.5 --node-name="cluster-1-node-1" --cluster-name="cluster-1" --network-host="127.0.0.1" --http-port=39210 --master-nodes="cluster-1-node-1" --seed-hosts="127.0.0.1:39300" --car="4gheap,trial-license" | jq --raw-output '.["installation-id"]'
esrally start --installation-id $ID --race-id 1


### Cluster 2
# install with basic license to fail
esrally install --quiet --distribution-version=7.17.5 --node-name="cluster-2-node-1" --cluster-name="cluster-1" --network-host="127.0.0.1" --http-port=39220 --master-nodes="cluster-1-node-1" --seed-hosts="127.0.0.1:39400" | jq --raw-output '.["installation-id"]'
esrally start --installation-id $ID --race-id 2

# install with trial license to succeeed
esrally install --quiet --distribution-version=7.17.5 --node-name="cluster-2-node-1" --cluster-name="cluster-1" --network-host="127.0.0.1" --http-port=39220 --master-nodes="cluster-1-node-1" --seed-hosts="127.0.0.1:39400" --car="4gheap,trial-license" | jq --raw-output '.["installation-id"]'
esrally start --installation-id $ID --race-id 2


esrally race --pipeline=benchmark-only --target-hosts='{"default":["127.0.0.1:39210"],"remote":["127.0.0.1:39220"]}' --track-path="/Users/bradleydeam/perf/github.com/b-deam/rally-tracks/elastic/logs" --challenge="cross-clusters-search" --car="4gheap,trial-license,x-pack-security" --client-options='{"default": {"timeout": 20,"use_ssl":false},"remote":{"timeout": 20,"use_ssl":false}}'  --track-params="number_of_replicas:0" --test-mode --kill-running-processes
```



Example error on failed remote connection:
```
    ____        ____
   / __ \____ _/ / /_  __
  / /_/ / __ `/ / / / / /
 / _, _/ /_/ / / / /_/ /
/_/ |_|\__,_/_/_/\__, /
                /____/

[INFO] Race id is [ee54825d-f6e9-4d9c-a534-d786cfa6cace]
[INFO] Racing on track [logs], challenge [cross-clusters-search] and car ['external'] with version [7.17.4].

Running insert-pipelines                                                       [100% done]
Running insert-ilm                                                             [100% done]
Running delete-all-datastreams                                                 [100% done]
Running delete-all-composable-templates                                        [100% done]
Running delete-all-component-templates                                         [100% done]
Running create-all-component-templates                                         [100% done]
Running create-all-composable-templates                                        [100% done]
Running create-required-data-streams                                           [100% done]
Running check-cluster-health                                                   [100% done]
Running wait-until-merges-finish                                               [100% done]
Running bulk-index                                                             [100% done]
Running connect-local-cluster-to-remote-cluster                                [  0% done]
[ERROR] Cannot race. Error in load generator [0]
        Cannot run task [connect-local-cluster-to-remote-cluster]: Unable to connect [default] to cluster [remote]

Getting further help:
*********************
* Check the log files in /Users/bradleydeam/.rally/logs for errors.
* Read the documentation at https://esrally.readthedocs.io/en/latest/.
* Ask a question on the forum at https://discuss.elastic.co/tags/c/elastic-stack/elasticsearch/rally.
* Raise an issue at https://github.com/elastic/rally/issues and include the log files in /Users/bradleydeam/.rally/logs.
```

File: elastic/shared/parameter_sources/workflow_selector.py [hunk 1]
```diff
@@ -70,7 +70,9 @@ def __init__(self, track, params, **kwargs):
         self._detailed_results = track.selected_challenge_or_default.parameters.get(
             "detailed-results", False
         )
-        self._workflow_target = track.selected_challenge_or_default.parameters.get("workflow-target")
+        self._workflow_target = params.get(
+            "workflow-target",
```
Comment (DJRickyB) [inline elastic/shared/parameter_sources/workflow_selector.py:74]: 👍 cool to add the task-local override here

**Review by DJRickyB — CHANGES_REQUESTED** [review]
No good deed goes unpunished. Can you add tests? The easier one should be adding a case to `workflow_selector_test.py` for task-level `workflow-target` taking precedence over a track parameter. There aren't a lot of bells and whistles to your runners here so I'll leave it to you to determine what's appropriate to do there, possibly extracting out some of the code to testable functions

**Review by DJRickyB — APPROVED** [review]
Nice work, this LGTM

**Comment by b-deam** [comment]
I added exception handling for the follow request, as it could fail silently otherwise.

( I also inadvertently found a Rally bug in how we check whether a runner is multi cluster aware, but it's not a blocker for this track. )

**Pull request merged**